src/libs/antares/study/parameters.cpp

```diff
@@ -691,327 +791,169 @@ static bool SGDIntLoadFamily_P(Parameters& d, const String& key, const String& v
+
     // deprecated
     if (key == "thresholdmin")
         return true; // value.to<int>(d.thresholdMinimum);
     if (key == "thresholdmax")
         return true; // value.to<int>(d.thresholdMaximum);
-    // Error
+
     return false;
 }
 
-static bool SGDIntLoadFamily_U(Parameters& d, const String& key, const String& value, uint)
-{
-    if (key == "user-playlist")
-        return value.to<bool>(d.userPlaylist);
 
-    if (key == "unit-commitment-mode") // after 5.0
-    {
-        auto ucommitment = StringToUnitCommitmentMode(value);
-        if (ucommitment != ucUnknown)
-        {
-            d.unitCommitment.ucMode = ucommitment;
-            return true;
-        }
-        logs.warning() << "parameters: invalid unit commitment mode. Got '" << value
-                       << "'. reset to fast mode";
-        d.unitCommitment.ucMode = ucHeuristic;
-        return false;
-    }
 
-    // Error
-    return false;
-}
 
-static bool SGDIntLoadFamily_Y(Parameters& d, const String& key, const String& value, uint)
-{
-    if (key == "year-by-year")
-        return value.to<bool>(d.yearByYear);
-    // Error
-    return false;
-}
+
+
+
+
+
+
+
+
+
+
 
 static bool ParametersInternalLoadProperty(Parameters& params,
+                                           const String& sectionName,
                                            const String& key,
                                            const String& value,
                                            const String& rawvalue,
                                            uint version)
 {
-    char c = key.first();
-    if (c < 'a' || c > 'z')
+    char cKey = key.first();
+    char cSection = sectionName.first();
+    if (cKey < 'a' || cKey > 'z' || cSection < 'a' || cSection > 'z')
         return false;
 
-    // Jump table
-    // Do not forget the variable `key` and `value` are identical to
-    // `p->key` and `p->value` except they are already in the lower case format
-    typedef bool (*Callback)(Parameters&, const String&, const String&, uint);
-    static Callback jumper[] = {
-      &SGDIntLoadFamily_A,
-      nullptr,
-      &SGDIntLoadFamily_C,
-      &SGDIntLoadFamily_D,
-      nullptr,
-      &SGDIntLoadFamily_F,
-      &SGDIntLoadFamily_G,
-      nullptr,
-      &SGDIntLoadFamily_I,
-      &SGDIntLoadFamily_J,
-      nullptr,
-      &SGDIntLoadFamily_L,
-      &SGDIntLoadFamily_M,
-      &SGDIntLoadFamily_N,
-      nullptr,
-      &SGDIntLoadFamily_P,
-      nullptr,
-      &SGDIntLoadFamily_R,
-      &SGDIntLoadFamily_S,
-      &SGDIntLoadFamily_T,
-      &SGDIntLoadFamily_U,
-      nullptr,
-      nullptr,
-      nullptr,
-      &SGDIntLoadFamily_Y,
-      nullptr,
-    };
-    if (c == 'h')
-    {
-        // special prototype for H
-        return SGDIntLoadFamily_H(params, key, value, rawvalue, version);
-    }
-    else
-    {
-        int i = c - 'a';
-        if (jumper[i])
-            return (*(jumper[i]))(params, key, value, version);
-    }
-    return false;
+
+
```

Comment:
Rename this function to `checkKeySection(const String& s)` to check that the 1st character is in range a-z.

Call it whenever a section/property name is read.

Reply:
Partially corrected here : fa68674. 
I've made the correction only for parameters name (but not for section). Indeed, section name are already checked in the map and I'm not sure that verifying it before will make any difference in the performances of the app. 
I think that this control is useless, even for parameters, with the new implementation (which is not based on the first letter). You may want to remove this function.

Reply:
I agree